Tools/autotest/logger_metadata/emit_xml.py

```diff
@@ -3,47 +3,114 @@
 from __future__ import print_function
 
 import emitter
+from xml.sax.saxutils import quoteattr
+
+INDENT2 = "  "
+INDENT4 = "    "
+INDENT6 = "      "
+INDENT8 = "        "
+INDENT10 = "          "
+
 
 class XMLEmitter(emitter.Emitter):
     def preface(self):
         return """<?xml version="1.0" encoding="utf-8"?>
 <!-- Dynamically generated list of documented logfile messages (generated by parse.py) -->
-<loggermessagefile>
-"""
+<loggermessagefile>"""
+
     def postface(self):
         return "</loggermessagefile>"
 
     def start(self):
+        self.logname = "LogMessages.xml"
         self.fh = open("LogMessages.xml", mode='w')
         print(self.preface(), file=self.fh)
 
     def emit(self, doccos):
         self.start()
         for docco in doccos:
-            print('    <logformat name="%s">' % docco.name, file=self.fh)
+            print(INDENT2 + '<logformat name="%s">' % docco.name, file=self.fh)
             if docco.url is not None:
-                print('        <url>%s</url>' % docco.url, file=self.fh)
+                print(INDENT4 + '<url>%s</url>' % docco.url, file=self.fh)
             if docco.description is not None:
-                print('        <description>%s</description>' %
+                print(INDENT4 + '<description>%s</description>' %
                       docco.description, file=self.fh)
-            print('        <fields>', file=self.fh)
+            print(INDENT4 + '<fields>', file=self.fh)
             for f in docco.fields_order:
-                print('            <field name="%s">' % f, file=self.fh)
+                print(INDENT6 + '<field name="%s">' % f, file=self.fh)
                 if "description" in docco.fields[f]:
-                    print('                <description>%s</description>' %
-                          docco.fields[f]["description"], file=self.fh)
+                    print(INDENT8 + '<description>%s</description>' %
+                          quoteattr(docco.fields[f]["description"], {'>': "&gt;"}), file=self.fh)
```

Comment:
I don't understand why we're not always escaping all the same things in quoteattr.  Seems as if we should have our own wrapper around quoteattr which always passes in the extra stuff.

I don't understand why we're passing '>' in here as it appears to be part of the defaults.

Reply:
I get rid of this madness and make a real xml generation